lib/blacklight/catalog.rb

```diff
@@ -79,8 +79,11 @@ def track
 
     # displays values and pagination links for a single facet field
     def facet
+      
+      extra_controller_params = (params[:"facet.prefix"] ? {"facet.prefix"=>params[:"facet.prefix"]} : {})
```

Comment:
Can we push this logic down into the `Blacklight::SearchHelper` or part of `Blacklight::Solr::RequestBuilder`?

https://github.com/projectblacklight/blacklight/blob/6abc7e99c28d93490109acac47b0b63623901ca7/lib/blacklight/search_helper.rb#L94